modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/PartitionPruningMetadataTest.java

```diff
@@ -100,11 +101,76 @@ public class PartitionPruningMetadataTest extends AbstractPlannerTest {
             .distribution(IgniteDistributions.affinity(List.of(0), 1, 2))
             .build());
 
-    /** Basic test cases for partition pruning metadata extractor. */
-    @ParameterizedTest
+    /** Basic test cases for partition pruning metadata extractor, select case. */
+    @ParameterizedTest(name = "SELECT: {0}")
     @EnumSource(TestCaseBasic.class)
-    public void testBasic(TestCaseBasic testCaseSimple) {
-        checkPruningMetadata(testCaseSimple.data);
+    public void testBasicSelect(TestCaseBasic testCaseSimple) {
+        checkPruningMetadata(testCaseSimple.data, SqlKind.SELECT);
+    }
+
+    /** Basic test cases for partition pruning metadata extractor, delete case. */
+    @ParameterizedTest(name = "DELETE: {0}")
+    @EnumSource(TestCaseBasic.class)
+    public void testBasicDelete(TestCaseBasic testCaseSimple) {
+        checkPruningMetadata(testCaseSimple.data, SqlKind.DELETE);
+    }
+
+    /** Basic test cases for partition pruning metadata extractor, delete case. */
+    @ParameterizedTest(name = "INSERT: {0}")
+    @EnumSource(TestCaseBasicInsert.class)
+    public void testBasicInsert(TestCaseBasicInsert testCaseSimple) {
+        checkPruningMetadata(testCaseSimple.data, SqlKind.INSERT);
+    }
+
+    /** Basic test cases for partition pruning metadata extractor, delete case. */
+    @ParameterizedTest(name = "UPDATE: {0}")
+    @EnumSource(TestCaseBasicUpdate.class)
+    public void testBasicInsert(TestCaseBasicUpdate testCaseSimple) {
```

Comment:
we have to decided whether this test covers DELETE, UPDATE, or INSERT case

Reply:
fixed